### PR TITLE
Enable 'accept' attribute in 'input' label.

### LINKED ISSUE #1768: <input type="file"> ignores accept attribute
I notice that PR #777 added support for input elements with type "file" (and "color").

However, the `accept` attribute is ignored.

For example,

```
<input type="file" accept=".txt">
```

Will add a "Text Document" option to the list of file types in the Open dialog if you simply run this in a browser.

If you run it under Electron, it does not appear.

Tested only under Windows (8.1).

### PR DESCRIPTION
**Pull request by hokein**
Fixes #1768, #1780.

### CODE REVIEW

File: atom/browser/web_dialog_helper.cc [hunk 1]
```diff
@@ -56,10 +108,13 @@ void WebDialogHelper::RunFileChooser(content::WebContents* web_contents,
     }
 
     std::vector<base::FilePath> paths;
+    auto browser_context = AtomBrowserMainParts::Get()->browser_context();
```
Comment (zcbenz) [inline atom/browser/web_dialog_helper.cc]: We should use the browser_context of current WebContents.

Comment (hokein) [inline atom/browser/web_dialog_helper.cc]: The current implementation relies on the `PrefService` in `brightray::BrowserContext`, which is aligned with Chromium's. And `content::BrowserContext` doesn't has such interface.

I know current implementation will cause `<input>` using the last used directory even in different WebContents, but this behavior is same with Chromium's(Chromium will share this last used directory to all WebContents in the same `Profile`).

Comment (zcbenz) [inline atom/browser/web_dialog_helper.cc]: > The current implementation relies on the PrefService in brightray::BrowserContext, which is aligned with Chromium's. And content::BrowserContext doesn't has such interface.

We can get it from current window.

> I know current implementation will cause <input> using the last used directory even in different WebContents, but this behavior is same with Chromium's(Chromium will share this last used directory to all WebContents in the same Profile).

Chrome's `Profile` is equivalent to Electron's `Session`, both of them inherit from `content::BrowserContext`.

Comment (hokein) [inline atom/browser/web_dialog_helper.cc]: > We can get it from current window.

I didn't find any APIs in `NativeWindow`. :cry:

Comment (zcbenz) [inline atom/browser/web_dialog_helper.cc]: `NativeWindow` is a WebContentsObserver so you can call `web_contents()`.

Comment (zcbenz) [inline atom/browser/web_dialog_helper.cc]: Actually all `content::BrowserContext` created in Electron should be an `AtomBrowserContext`, it is safe to call `static_cast`.

Comment (hokein) [inline atom/browser/web_dialog_helper.cc]: Done.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**